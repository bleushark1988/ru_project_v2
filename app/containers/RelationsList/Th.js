import styled from 'styled-components';

const Th = styled.th`
  cursor: auto;
  background: #f9fafb;
  text-align: inherit;
  color: rgba(0, 0, 0, 0.87);
  padding: 0.92857143em 0.78571429em;
  vertical-align: inherit;
  font-weight: 700;
  text-transform: none;
  border-bottom: 1px solid rgba(34, 36, 38, 0.1);
  border-left: 1px solid rgba(34, 36, 38, 0.1);
  &:first-child {
    border-radius: 0.28rem;
    border-left: none;
  }
  &:last-child {
    border-radius: 0 0.28571429rem 0 0;
  }
`;

export default Th;
