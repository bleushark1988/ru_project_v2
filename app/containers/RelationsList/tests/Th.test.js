import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Th from '../Th';

describe('<Th />', () => {
  it('should match the snapshot', () => {
    const renderedComponent = renderer.create(<Th />).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should have a className attribute', () => {
    const renderedComponent = shallow(<Th />);
    expect(renderedComponent.prop('className')).toBeDefined();
  });

  it('should adopt a valid attribute', () => {
    const id = 'test';
    const renderedComponent = shallow(<Th id={id} />);
    expect(renderedComponent.prop('id')).toEqual(id);
  });

  it('should not adopt an invalid attribute', () => {
    const renderedComponent = shallow(<Th attribute="test" />);
    expect(renderedComponent.prop('attribute')).toBeUndefined();
  });
});
