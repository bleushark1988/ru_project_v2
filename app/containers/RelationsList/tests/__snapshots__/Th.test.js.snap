// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Th /> should match the snapshot 1`] = `
.c0 {
  cursor: auto;
  background: #f9fafb;
  text-align: inherit;
  color: rgba(0,0,0,0.87);
  padding: 0.92857143em 0.78571429em;
  vertical-align: inherit;
  font-weight: 700;
  text-transform: none;
  border-bottom: 1px solid rgba(34,36,38,0.1);
  border-left: 1px solid rgba(34,36,38,0.1);
}

.c0:first-child {
  border-radius: 0.28rem;
  border-left: none;
}

.c0:last-child {
  border-radius: 0 0.28571429rem 0 0;
}

<th
  className="c0"
/>
`;
